neutrino.go

```diff
@@ -1257,15 +1257,17 @@ func (s *ChainService) handleDonePeerMsg(state *peerState, sp *ServerPeer) {
 			s.connManager.Disconnect(sp.connReq.ID())
 		} else {
 			s.connManager.Remove(sp.connReq.ID())
-			go s.connManager.NewConnReq()
 		}
 		log.Debugf("Removed peer %s", sp)
-		return
+	} else {
+		// We'll always remove peers that are not persistent.
```

Comment:
Wasn't this always already hit above (outbound non-persistent). This is for peers that never completed the handshake, so were never added to our internal peer tracking state.

Reply:
Correct. I've unified the handling of peers being tracked and not being tracked to hopefully make this easier to follow.